src/store/channels-list/saga.test.ts

```diff
@@ -1,13 +1,31 @@
 import { expectSaga } from 'redux-saga-test-plan';
 import * as matchers from 'redux-saga-test-plan/matchers';
 
-import { api } from './api';
+import { fetchChannels } from './api';
 import { fetch } from './saga';
 
 import { setStatus } from '.';
 import { rootReducer } from '..';
 import { AsyncListStatus } from '../normalized';
 
+const MOCK_CHANNELS = [
+  { name: 'channel 1' },
+  { name: 'channel 2' },
+  { name: 'channel 3' },
+];
+const unmockedFetch = global.fetch;
+
+beforeAll(() => {
+  global.fetch = () =>
+    Promise.resolve({
+      json: () => Promise.resolve(MOCK_CHANNELS),
+    });
+});
+
+afterAll(() => {
+  global.fetch = unmockedFetch;
+});
+
```

Comment:
you shouldn't need to mock the global fetch like this. a good rule of thumb is if you find yourself mocking a global or an entire module, there's probably a better way to do it. feel free to reach out any time to ask. :)

react-saga-test-plan has mocking built-in. that's what the `provide` method does. so, for the first test below, all you have to do is: 
```
.provide([
        [
          matchers.call.fn(fetchChannels),
          MOCK_CHANNELS,
        ],
      ])
```